### ISSUE TITLE
Path negotiation commands

### ISSUE THREAD

**Issue by jackzampolin** [cosmos/relayer#26]
This command will look for existing clients, connections, and channels that link any pair of chains on the relayer and create a `Path` or `Path`s in the config to mirror that. Optionally, if some primitives don't exist, then names for them are generated and those primitives are created. This may be a number of different commands as the number of options here will grow quickly.

**Comment by jackzampolin** [comment]
Blocked on: https://github.com/cosmos/cosmos-sdk/issues/5829
Blocked on: https://github.com/cosmos/cosmos-sdk/issues/5617

**Comment by ebuchman** [comment]
So I managed to pull this off manually as follows:

- `rly query clients <src-chain>`, find the entry with the chain_id of the `<dst-chain>`, and grab that client id. now we have the `<src-client-id>`
- `rly query client-connections <src-chain> <src-client-id>` to get the connection id. now we have the `<src-conn-id>`.
- `rly query connection <src-chain> <src-conn-id>` to get the destination chains client id and connection id. now we have `<dest-client-id>` and `<dest-conn-id>`
- `rly query channels <src-chain>`, find the one that has the `<src-conn-id>` in the `connection_hops`, and from there get the destination channel id. 
- `rly query channels <dst-chain>`, find the one that has the `<dst-conn-id>` in the `connection_hops`, and from there get the source channel id

Now we have the client/conn/channel ids for both src and dst and can make the path manually. Whew!

Obviously it would be amazing if one command could just get us what we wanted, but in the meantime this "works". One thing I found weird though is I can't find channel ids directly on the chain, only the counterparty channel id. Is that blocked on something upstream, or can we fix it here?

Given we can get all this, we could consider implementing it as a command that just does multiple queries as necessary to get the data we want to compose the path, and then possibly don't need to block on the upstream changes? Not sure how difficult those upstream changes will be.

**Comment by jackzampolin** [comment]
> Obviously it would be amazing if one command could just get us what we wanted, but in the meantime this "works". One thing I found weird though is I can't find channel ids directly on the chain, only the counterparty channel id. Is that blocked on something upstream, or can we fix it here?

See the linked issues.

**Comment by jackzampolin** [comment]
>Given we can get all this, we could consider implementing it as a command that just does multiple queries as necessary to get the data we want to compose the path, and then possibly don't need to block on the upstream changes? Not sure how difficult those upstream changes will be.

Should be pretty easy. Should have this working mid-week next week.

**Comment by jackzampolin** [comment]
And this has taken a bit longer than expected but I have a good deal of this functionality in the relayer. Once #104 lands `rly paths gen` will try to reuse any existing primitives automagically. Previously, I've already added an experimental `rly paths find` command that will find all paths between all configured chains. It works pretty well on the 4 node testnet I've been working with locally.

This functionality should be developed further. I will leave this issue open for that work.

**Comment by jackzampolin** [comment]
Closing in favor of new issue about path command.
